python/src/lib.rs

```diff
@@ -302,16 +361,18 @@ impl RawDeltaTable {
         retention_hours: Option<u64>,
         enforce_retention_duration: bool,
     ) -> PyResult<Vec<String>> {
-        let mut cmd = VacuumBuilder::new(self._table.object_store(), self._table.state.clone())
+        let mut table = self._rt.block_on(self._table.lock());
+        let mut cmd = VacuumBuilder::new(table.object_store(), table.state.clone())
```

Comment:
is there a way for us to not clone the state? Since are are consuming it in the builder, and replacing the table object later we are actually not re-using the current state anywhere, right?

The same applies to other operations.